File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.js
var largestInteger = function(nums, k) {
    const n = nums.length;
    const count = new Array(51).fill(0);

    for (let i = 0; i <= n - k; i++) {
        const seen = new Set();

        for (let j = i; j < i + k; j++) {
            seen.add(nums[j]);
        }

        for (const x of seen) {
            count[x]++;
        }
    }

    for (let x = 50; x >= 0; x--) {
        if (count[x] === 1) {
            return x;
        }
    }

    return -1;
};

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.cs
public class Solution {
    public int LargestInteger(int[] nums, int k) {
        int n = nums.Length;
        int[] count = new int[51];

        for (int i = 0; i <= n - k; i++) {
            HashSet<int> seen = new HashSet<int>();

            for (int j = i; j < i + k; j++) {
                seen.Add(nums[j]);
            }

            foreach (int x in seen) {
                count[x]++;
            }
        }

        for (int x = 50; x >= 0; x--) {
            if (count[x] == 1) {
                return x;
            }
        }

        return -1;
    }
}